Observer/TescoPaymentObserver.php

```diff
@@ -46,7 +46,7 @@ public function execute(Observer $observer)
 
         $paymentData = $payment->getData();
 
-        if ($paymentData['additional_information']['payment_type'] !== 'bill_payment_tesco_lotus') {
+        if (!isset($paymentData['additional_information']) || !isset($paymentData['additional_information']['payment_type']) || $paymentData['additional_information']['payment_type'] !== 'bill_payment_tesco_lotus') {
```

Comment:
@jacstn I just looked at `module-sales/Model/Order/Payment.php` and ``module-sales/Model/Order/Payment/Info.php`. 
Seems we can use 👇 instead.
```php
if (! $payment->hasAdditionalInformation('payment_type') || $payment->getAdditionalInformation('payment_type') !== 'bill_payment_tesco_lotus ') {
    // ...
}
```

Or could even be
```php
if ($payment->getAdditionalInformation('payment_type') !== 'bill_payment_tesco_lotus ') {
    // ...
}
```
`getAdditionalInformation($key)` will return `null` if the key doesn't exist.

Reply:
yes.. but that `$payment->getAdditionalInformation('barcode')`
returned always null for some reason.

Reply:
Oh, it might be because it contains URL?
Maybe escape HTML once before set to additionalInformation would help?

But well, that's another topic.
For now, I think that `!isset($paymentData['additional_information'])` is unncessary but if you want to have it here, then you might want to consider to have it at `Block/Checkout/Onepage/Success/TescoAdditionalInformation.php` as well? 

Cuz in that class it also checks similar thing but just only
```php
if (!isset($paymentData['additional_information']['payment_type']) || $paymentData['additional_information']['payment_type'] !== 'bill_payment_tesco_lotus') {
    // ...
}
```

Reply:
the barcode is SVG raw code... but your suggestion might be a good lead.

also, I wanted to be consistant.. in the file.. it is only one way to approach, it might look better, to reduce code, but it is only one way to do it.

since I will have to get that `additional_data` in a raw array to get a barcode, then why not to do it all in the same way (treat additional data).

I will try to get that `barcode` using that method, will try to escape string.

Reply:
I mean I think that `!isset($paymentData['additional_information'])` is unnecessary to check cuz after looked at Magento classes, `additional_information` is a default key generated by Magento, so it's should always presented there.

But if you want to check it, then maybe we should also update `Block/Checkout/Onepage/Success/TescoAdditionalInformation.php` class, cuz right now that class does not check for `!isset($paymentData['additional_information'])` case.

Reply:
ok.. yes.. that might be too much